### PR TITLE
Issue #28032: VAT Tax Handling on (misc) Cash Receipts

### PR DESCRIPTION
**Pull request by andersdd**
plus some cash receipt bugfixes from the Customer Group feature

### CODE REVIEW

**Comment by andersdd** [comment]
@gilmoskowitz has something updated/changed with jshint?  This test failure looks like a whole bunch of node or core files unrelated to the sql files I have changed??

**Comment by gilmoskowitz** [comment]
@andersdd I don't know what's broken. As far as I can tell, nothing has changed with jshint yet. My changes in #2714 and the related changes I'm working on in xtuple/private-extensions will. Those have not been completed yet.

There's something suspicious about the list of files generating complaints. None of them appear to be ours and should have been skipped. I'll look more closely after the CI rerun completes.

**Comment by gilmoskowitz** [comment]
@andersdd - https://travis-ci.org/xtuple/xtuple/builds/137385525#L3348
```
2016-06-13 23:53:49 UTC ERROR:  "_taxaccnt" is not a known variable at character 859
2016-06-13 23:53:49 UTC STATEMENT:  CREATE OR REPLACE FUNCTION _cashRcptMiscTrigger () RETURNS TRIGGER AS $$
```

File: foundation-database/public/trigger_functions/cashrcptmisc.sql [hunk 1]
```diff
@@ -8,32 +8,40 @@ BEGIN
 
   -- Checks
   -- Start with Privileges
-  IF (TG_OP = ''INSERT'') THEN
-    SELECT checkPrivilege(''MaintainCashReceipts'') INTO _check;
+  IF (TG_OP = 'INSERT') THEN
+    SELECT checkPrivilege('MaintainCashReceipts') INTO _check;
     IF NOT (_check) THEN
-      RAISE EXCEPTION ''You do not have privileges to add a new Cash Receipt Misc. Application.'';
+      RAISE EXCEPTION 'You do not have privileges to add a new Cash Receipt Misc. Application.';
```
Comment (gilmoskowitz) [inline foundation-database/public/trigger_functions/cashrcptmisc.sql]: The message should include the `[xtuple: funcname, errcode]` suffix.

**Pull request merged**